etchosts/etchosts.go

```diff
@@ -119,25 +119,19 @@ func Add(path string, recs []Record) error {
 		return nil
 	}
 
-	f, err := os.Open(path)
-	if err != nil {
-		return err
-	}
-
-	content := bytes.NewBuffer(nil)
-
-	_, err = content.ReadFrom(f)
+	f, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND, 0644)
 	if err != nil {
 		return err
 	}
+	defer f.Close()
 
 	for _, r := range recs {
-		if _, err := r.WriteTo(content); err != nil {
```

Comment:
storing into a buffer and writing all at once is not giving atomicity. I understand what you are trying to do and am working on another change which will use https://godoc.org/github.com/docker/docker/pkg/ioutils#NewAtomicFileWriter, but did not want to make that larger update here since it requires a vendor update. If you guys don't want to change this then I can just add a close for now and address atomicity later.